core/src/avm1/globals.rs

```diff
@@ -55,6 +55,9 @@ pub fn create_globals<'gc>(gc_context: MutationContext<'gc, '_>) -> Object<'gc>
     globals.force_set("Math", Value::Object(math::create(gc_context)));
     globals.force_set_function("getURL", getURL, gc_context);
     globals.force_set_function("random", random, gc_context);
+    globals.force_set("NaN", Value::Number(std::f64::NAN));
+    globals.force_set("Infinity", Value::Number(std::f64::INFINITY));
+    globals.force_set("-Infinity", Value::Number(std::f64::NEG_INFINITY));
```

Comment:
I don't think `-Infinity` actually exists, it's just `Infinity` followed by a subtract from 0 op

Reply:
I _think_ you're right and I've pushed that change but I just want to double check that. I was seeing it in the constant pool but most likely because I am testing `"-Infinity"` too. The syntax highlighting in flash turns the `-` blue with the word `Infinity`. It doesn't do that for `-NaN`. So hold off on this and I will just confirm for certain by creating a swf with just `typeof -Infinity` in it and checking the opcodes.

Reply:
Another way to check is
```
trace(_global["Infinity"]);
trace(_global["-Infinity"]);
```

Funny that you can overwrite `NaN`/`Infinity` , I didn't know that!